Source/DafnyLanguageServer.Test/Various/PluginsAdvancedTest.cs

```diff
@@ -0,0 +1,168 @@
+    return new Rewriter[]{new ExternCheckRewriter(errorReporter, this)};
+  }
+}
+
+public class ExternCheckRewriter: Rewriter {
+  private readonly TestConfiguration configuration;
+
+  public ExternCheckRewriter(ErrorReporter reporter, TestConfiguration configuration): base(reporter) {
+    this.configuration = configuration;
+  }
+
+  public override void PostResolve(Program program) {
+    foreach (var moduleDefinition in program.Modules()) {
+      foreach (var topLevelDecl in moduleDefinition.TopLevelDecls) {
+        if (topLevelDecl is ClassDecl cd) {
+          foreach (var member in cd.Members) {
+            if (member is Method methodExtern) {
+              if (Attributes.Contains(member.Attributes, ""extern"")) {
+                // Verify that there exists a test method which references this extern method.
+                var tested = false;
+                Method candidate = null;
+                foreach (var member2 in cd.Members) {
+                  if (member2 == member || !(member2 is Method methodTest)) {
+                    continue;
+                  }
+                  if (!Attributes.Contains(methodTest.Attributes, ""test"")) {
+                    continue;
+                  }
+                  if (!moduleDefinition.CallGraph.Reaches(methodTest, methodExtern)) {
+                    continue;
+                  }
+                  candidate = methodTest;
+
+                  if (configuration.ForceName && candidate.Name != methodExtern.Name + ""_test"") {
+                    continue;
+                  }
+                  tested = true;
+                  break;
+                }
+
+                if (!tested) {
+                  var forceMessage = configuration.ForceName ? $"" named {methodExtern.Name}_test"" : """";
+                  var token = configuration.ForceName && candidate != null
+                    ? new NestedToken(methodExtern.tok, candidate.tok, ""You might want to just rename this method"")
+                    : methodExtern.tok;
+                  Reporter.Error(MessageSource.Resolver, token,
+                    $""Please declare a method {{:test}}{forceMessage} that will call {methodExtern.Name}{configuration.PluginUser}"");
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}");
+    client = await InitializeClient(options => options.OnPublishDiagnostics(diagnosticReceiver.NotificationReceived));
+  }
+
+  protected override IConfiguration CreateConfiguration() {
+    return new ConfigurationBuilder().AddCommandLine(
+      new[] { $@"--dafny:plugins:0=""{libraryPath},force you""" }).Build();
+  }
+
+  /// <summary>
+  /// This method creates a library and returns the path to that library.
+  /// The library extends a Rewriter so that we can verify that Dafny invokes it if provided in argument.
+  /// </summary>
+  public async Task<string> GetLibrary(string code) {
```

Comment:
The compiler already states it. Either change this method to be truly async by awaiting some statements (I think some of the used Roslyn APIs do have async counterparts), or change the signature to be purely synchronous. I'd opt for the former, if there are suitable APIs.

Reply:
I did not find suitable APIs to do this asynchronously. I think it's fine to do it synchronously, so I changed the code to reflect that